tests/testthat/test-lint_package.R

```diff
@@ -132,3 +132,43 @@ test_that("lint_package returns early if no package is found", {
     fixed = TRUE
   )
 })
+
+test_that(
+  "`lint_package` will use a `.lintr` file in `.github/linters/` directory the same as the package root",
+  {
+    withr::local_options(lintr.linter_file = "lintr_test_config")
+
+    pkg_path <- test_path("dummy_packages", "github_lintr_file")
+
+    dir.create(
+      path = file.path(pkg_path, ".github/linters/"),
+      recursive = TRUE
+    )
+
+    # In `github/linters`add a `.lintr` file that excludes the whole of `abc.R`
+    # and the first line of `jkl.R` (and remove it on finishing this test)
+    local_config(file.path(pkg_path, ".github/linters"), "exclusions: list('R/abc.R', 'R/jkl.R' = 1)")
+
+    lints_using_github_lintr <- withr::with_dir(
+      pkg_path,
+      lint_package(".", linters = list(assignment_linter()))
+    )
+
+    # Now add the same file to the package root
```

Comment:
Delete the `.github/linters/.lintr` file here for more safety

Reply:
Code updated, marking as resolved